### PR TITLE
[fluorine] Merge forward from 2018.3 to fluorine

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - tests/integration/states/test_file.py

### CODE REVIEW

**Review by gtmanfred — APPROVED** [review]

File: conf/master [hunk 1]
```diff
@@ -125,7 +125,7 @@
 # The master can take a while to start up when lspci and/or dmidecode is used
 # to populate the grains for the master. Enable if you want to see GPU hardware
 # data for your master.
-# enable_gpu_grains: False
+# enable_gpu_grains: True
```
Comment (terminalmage) [inline conf/master]: This is definitely wrong, but was broken back in 2017.7 about a week ago. In fact, everything about 2fe675cfdd86ba370f3eec70dd50d149a70d04e5 is incorrect with the exception of the change in `doc/ref/configuration/master.rst`. I've opened https://github.com/saltstack/salt/pull/49375 separately to fix this.

**Review by terminalmage — APPROVED** [review]
Approved with a caveat that I've already separately fixed.

**Comment by gtmanfred** [comment]
This is currently failing on multiple test runs

```
integration.states.test_git.LocalRepoGitTest.test_latest_force_reset_true_fast_forward
```

**Comment by rallytime** [comment]
Okie doke. I will fix that first thing tomorrow.

**Pull request merged**